phoenix-core/src/main/java/org/apache/phoenix/coprocessor/UngroupedAggregateRegionObserver.java

```diff
@@ -1146,6 +1154,45 @@ private void setMutationAttributes(Mutation m, byte[] uuidValue) {
             m.setDurability(Durability.SKIP_WAL);
         }
 
+        private Delete generateDeleteMarkers(List<Cell> row) {
+            Set<ColumnReference> allColumns = indexMaintainer.getAllColumns();
+            if (row.size() == allColumns.size() + 1) {
+                return null;
+            }
+            Set<ColumnReference> includedColumns = Sets.newLinkedHashSetWithExpectedSize(row.size());
+            long ts = 0;
+            for (Cell cell : row) {
+                includedColumns.add(new ColumnReference(CellUtil.cloneFamily(cell), CellUtil.cloneQualifier(cell)));
+                if (ts < cell.getTimestamp()) {
+                    ts = cell.getTimestamp();
+                }
+            }
```

Comment:
Lines from 1163 to 1169 seem to be a good candidate to have in a separate function

Reply:
They are already in a function, which is generateDeleteMarkers.